tests/sentry/tasks/test_post_process.py

```diff
@@ -1936,8 +1936,7 @@ def test_replay_linkage(self, incr, kafka_producer, kafka_publisher):
         assert ret_value["segment_id"] is None
         assert ret_value["retention_days"] == 90
 
-        # convert ret_value_payload which is a list of bytes to a string
-        ret_value_payload = json.loads(bytes(ret_value["payload"]).decode("utf-8"))
+        ret_value_payload = json.loads(bytes(ret_value["payload"]))
```

Comment:
this is... puzzling -- I think there's definitely something wrong with how this data is being transmitted:

```
(Pdb) ret_value['payload']
[123, 34, 116, 121, 112, 101, 34, 58, 34, 101, 118, 101, 110, 116, 95, 108, 105, 110, 107, 34, 44, 34, 114, 101, 112, 108, 97, 121, 95, 105, 100, 34, 58, 34, 49, 55, 57, 51, 55, 102, 99, 51, 54, 55, 55, 101, 52, 48, 48, 97, 56, 57, 49, 53, 53, 51, 101, 52, 100, 55, 54, 53, 101, 50, 98, 97, 34, 44, 34, 101, 118, 101, 110, 116, 95, 104, 97, 115, 104, 34, 58, 34, 55, 48, 100, 51, 102, 57, 54, 54, 45, 57, 50, 55, 55, 45, 99, 97, 98, 54, 45, 102, 54, 102, 54, 45, 98, 52, 57, 55, 100, 53, 51, 49, 99, 52, 53, 49, 34, 44, 34, 116, 105, 109, 101, 115, 116, 97, 109, 112, 34, 58, 49, 55, 49, 52, 52, 49, 55, 48, 51, 55, 44, 34, 101, 114, 114, 111, 114, 95, 105, 100, 34, 58, 34, 101, 50, 98, 102, 99, 50, 97, 50, 102, 49, 101, 57, 52, 50, 100, 54, 56, 48, 101, 48, 98, 97, 98, 52, 99, 54, 97, 101, 52, 101, 49, 57, 34, 125]
(Pdb) bytes(ret_value['payload'])
b'{"type":"event_link","replay_id":"17937fc3677e400a891553e4d765e2ba","event_hash":"70d3f966-9277-cab6-f6f6-b497d531c451","timestamp":1714417037,"error_id":"e2bfc2a2f1e942d680e0bab4c6ae4e19"}'
```

Reply:
@anonrig is handling this separately by fixing the underlying data structure here (instead of 4x bytes-per-json-byte encoding this!)